In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

In [6]:
raw_data = pd.read_csv('2.02. Binary predictors.csv')
raw_data.head()

SAT Admitted  Gender
0  1363       No    Male
1  1792      Yes  Female
2  1954      Yes  Female
3  1653       No    Male
4  1593       No    Male

In [8]:
data = raw_data.copy()
data['Admitted'] = data['Admitted'].map({'Yes':1,'No':0})
data['Gender'] = data['Gender'].map({'Female':1,'Male':0})
data.head()

SAT  Admitted  Gender
0  1363         0       0
1  1792         1       1
2  1954         1       1
3  1653         0       0
4  1593         0       0

In [15]:
x1 = data[['SAT','Gender']]
y = data['Admitted']

In [16]:
x = sm.add_constant(x1)
log_result = sm.Logit(y,x).fit()
log_result.summary()

Optimization terminated successfully.
         Current function value: 0.120117
         Iterations 10


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:               Admitted   No. Observations:                  168
Model:                          Logit   Df Residuals:                      165
Method:                           MLE   Df Model:                            2
Date:                Sat, 25 Jan 2020   Pseudo R-squ.:                  0.8249
Time:                        22:04:46   Log-Likelihood:                -20.180
converged:                       True   LL-Null:                       -115.26
Covariance Type:            nonrobust   LLR p-value:                 5.118e-42
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const        -68.3489     16.454     -4.154      0.000    -100.598     -36.100
SAT            0.0406      0.010      4.129      0.000       0.021       0.060
Gender         1.9449      0.846      2.299      0.022       0.287       3.603
==============================================================================

Possibly complete quasi-separation: A fraction 0.27 of observations can be
perfectly predicted. This might indicate that there is complete
quasi-separation. In this case some parameters will not be identified.
"""

In [17]:
np.exp(1.9449)

6.992932526814459

#### With the information given, though lacking context, we can say that goven the same SAT score, a female is 6.99 times more likely to get admitted